### PR TITLE
add try catch block to stop crash

### PR DESCRIPTION
**Pull request by prkay**
This MR is raised to fix https://github.com/ekasetiawans/flutter_background_service/issues/353
Please merge this PR

### CODE REVIEW

File: packages/flutter_background_service_android/android/src/main/java/id/flutter/flutter_background_service/WatchdogReceiver.java [hunk 1]
```diff
@@ -66,10 +66,14 @@ public void onReceive(Context context, Intent intent) {
...
-                } else {
-                    context.startService(new Intent(context, BackgroundService.class));
+                try {
+                    if (config.isForeground()) {
+                        ContextCompat.startForegroundService(context, new Intent(context, id.flutter.flutter_background_service.BackgroundService.class));
+                    } else {
+                        context.getApplicationContext().startService(new Intent(context, id.flutter.flutter_background_service.BackgroundService.class));
+                    }}
+                catch (Exception e){
+                    e.printStackTrace();
```
Comment (repolyo) [inline packages/flutter_background_service_android/android/src/main/java/id/flutter/flutter_background_service/WatchdogReceiver.java:76]: What are the exception/error we are trying to catch here and root cause?

File: packages/flutter_background_service/pubspec.yaml [hunk 2]
```diff
@@ -15,7 +15,11 @@ dependencies:
     sdk: flutter
 
   flutter_background_service_platform_interface: ^5.0.0
-  flutter_background_service_android: ^6.0.1
+  flutter_background_service_android:
+    git:
+      url: https://github.com/ekasetiawans/flutter_background_service.git
+      ref: 37200c0086ffd0284ffb18b033ef90e499536d28
+      path: packages/flutter_background_service_android
```
Comment (repolyo) [inline packages/flutter_background_service/pubspec.yaml:22]: latest version for flutter_background_service_android is v5.0.2.
What is the reason of not using that?
In other words, why use a specific git revision => '37200c0086ffd0284ffb18b033ef90e499536d28?

**Comment by ekasetiawans** [comment]
hi @prkay could you resolve the conflicts

File: packages/flutter_background_service_android/android/src/main/java/id/flutter/flutter_background_service/WatchdogReceiver.java (hunk 1 above)
Comment (prkay) [reply packages/flutter_background_service_android/android/src/main/java/id/flutter/flutter_background_service/WatchdogReceiver.java:76]: Root cause for this is line no. 73
When context.getApplicationContext() is called sometimes it returns null. Due to which crash is happening

**Comment by prkay** [comment]
> hi @prkay could you resolve the conflicts?

Hi @ekasetiawans 
I had resolved the conflicts. Can you please review this MR.

**Pull request merged**